common/src/main/java/com/wynntils/services/mapdata/attributes/MapAttributesBuilder.java

```diff
@@ -27,10 +27,11 @@ public class MapAttributesBuilder {
     private String icon;
     private MapVisibilityImpl iconVisibility;
     private CustomColor iconColor;
+    private Boolean hasMarker;
```

Comment:
Oh, I really don't like Boolean and Float. I guess the intention here is to allow null somehow? But that is just really confusing. What does a null value for `hasMarker` mean? I don't recall the details about attribute resolution, but we clearly manage to have `int` (and not `Integer`) for priority and level.

Reply:
These fields have to be the boxed nullable type because they are not present for all types of map features. I don't remember exactly, but maybe it causes the json sanity checking to fail (the check that ensures that only the valid fields are set) as GSON gives these fields a default value.

Reply:
Ok, fine. I'm prepared to lower my standards of understanding just to get this done at this point.